### PR TITLE
WIP: Add rudimentary LDAP group support

### PR DESCRIPTION
**Pull request by stertingen**
Targeting https://github.com/go-gitea/gitea/issues/2212.

### Features:
- Restrict automatic registration with a group filter
- Allow admin access based on a group filter
- Arbitrary kinds of LDAP groups are supported (custom filter with custom user attribute)

### Sub-tasks
- [x] Add configuration options for the admin backend
- [x] Extend login code with new group options
- [x] Tests for new options suitable for the test OpenLDAP server
- [ ] Test with another LDAP setup (not using dn as user attribute in group)
- [ ] Refactor LDAP code to remove duplication (esp. SearchEntry and SearchEntries)
- [ ] Add CLI options (see https://github.com/go-gitea/gitea/pull/6681)
- [ ] Test CLI options

### CODE REVIEW

**Comment by stertingen** [comment]
Well, I forgot to apply the changes to SearchEntries().

File: integrations/auth_ldap_test.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func addAuthSourceLDAP(t *testing.T, sshKeyAttribute string) {
 		"attribute_name":           "givenName",
 		"attribute_surname":        "sn",
 		"attribute_mail":           "mail",
-		"attribute_ssh_public_key": sshKeyAttribute,
+		"attribute_ssh_public_key": "sshPublicKey",
```
Comment (lafriks) [inline integrations/auth_ldap_test.go]: why this change? This was intentionally to have option to test different scenarious

File: options/locale/locale_de-DE.ini [hunk 2]
```diff
@@ -1646,6 +1646,12 @@ auths.use_paged_search=Seitensuche verwenden
 auths.search_page_size=Seitengröße
 auths.filter=Benutzerfilter
 auths.admin_filter=Admin-Filter
+auths.group_search_base = Basis für Gruppensuche
```
Comment (lafriks) [inline options/locale/locale_de-DE.ini]: Do not add translations as they will be lost anyway on sync with crowdin

File: integrations/auth_ldap_test.go (hunk 1 above)
Comment (stertingen) [reply integrations/auth_ldap_test.go]: Well, you're right. Even if it was unnecessary, it's out of scope for this PR.

**Comment by lunny** [comment]
CI is still failed and will you finish the left sub tasks on the comment?
